### ISSUE TITLE
error in `injected-web.js` if web page does `Object.prototype.set=function(){};`

### ISSUE THREAD

**Issue by xBZZZZ** [violentmonkey/violentmonkey#1418]
### How to reproduce it?
1. install this userscript:
```javascript
// ==UserScript==
// @name        useless script
// @namespace   http://bzzzzdzzzz.blogspot.com/
// @description does nothing
// @author      BZZZZ
// @include     *
// @version     0.1
// @grant       none
// @run-at      document-end
// ==/UserScript==

undefined;
```
2. host this on http server (make sure `Content-Type` is `text/html`):
```html
<script>
Object.prototype.set=function(){};
</script>
```
3. open it in browser
### What is the expected result?
nothing appears in console
### What is the actual result?
```
Uncaught (in promise) TypeError: Invalid property descriptor. Cannot both specify accessors and a value or writable attribute, #<Object>
    at defineProperty (<anonymous>)
    at ot (injected-web.js:1)
    at injected-web.js:1
    at Rt (injected-web.js:1)
    at r (injected-web.js:1)
    at Dt (injected-web.js:1)
    at set (injected-web.js:1)
    at useless script.user.js:1
```
appears in console
### Environment
- Browser: Chromium
- Violentmonkey version: Violentmonkey BETA v2.13.0.10

**Comment by tophf** [comment]
If the page wants to break userscripts there are hundreds of things it can break so even if Violentmonkey itself will run on such a page. Fixing this won't be useful because we can't protect the userscripts in `page` mode.

If someone encounters such a page we will suggest them to switch injection to `content` mode.

If we see more reports of real sites doing that, we will maintain a blacklist of such vandal sites and enable `content` mode automatically.

**Comment by tophf** [comment]
On the other hand, the underlying issue is interesting because evidently Object.defineProperty reads inherited properties, which is rather non-sensical in my opinion, so I'll fix this particular thing because it's well... interesting :-)

**Comment by tophf** [comment]
Events were also vulnerable, fixed in b075648.
